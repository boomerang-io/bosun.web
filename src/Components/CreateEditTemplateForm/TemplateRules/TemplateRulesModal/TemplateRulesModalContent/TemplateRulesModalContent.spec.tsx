import React from "react";
import Inputs from ".";
import { fireEvent, waitFor } from "@testing-library/react";

const mockfn = jest.fn();

const props = {
  isEdit: true,
  input: {
    defaultValue: "dogs",
    description: "Tim property",
    key: "tim.property",
    label: "Tim Property",
    required: false,
    type: "text",
  },
  loading: false,
  updateInputs: mockfn,
  closeModal: mockfn,
  inputsName: [],
  workflowActions: { updateWorkflowInput: mockfn, createWorkflowInput: mockfn },
};

describe("Inputs --- Snapshot Test", () => {
  it("Capturing Snapshot of Inputs", () => {
    const { baseElement } = rtlRender(<Inputs {...props} />);
    expect(baseElement).toMatchSnapshot();
  });
});

describe("Inputs --- RTL", () => {
  it("Change default value by type correctly", () => {
    const { getByText, getByPlaceholderText, queryByTestId } = rtlRender(<Inputs {...props} />);
    expect(queryByTestId("text-input")).toBeInTheDocument();

    const typeSelect = getByPlaceholderText(/select an item/i);

    fireEvent.click(typeSelect);
    fireEvent.click(getByText(/text area/i));

    expect(queryByTestId("toggle")).not.toBeInTheDocument();
    expect(queryByTestId("text-area")).toBeInTheDocument();

    fireEvent.click(typeSelect);
    fireEvent.click(getByText(/select/i));

    expect(queryByTestId("text-area")).not.toBeInTheDocument();
    expect(queryByTestId("select")).toBeInTheDocument();
  });

  it("Shouldn't save property without key and label", async () => {
    const { getByText, getByPlaceholderText, getByTestId } = rtlRender(
      <Inputs {...props} isEdit={false} input={undefined} />
    );
    fireEvent.click(getByText(/create/i));
    await waitFor(() => expect(getByTestId("inputs-modal-confirm-button")).toBeDisabled());

    const keyInput = getByPlaceholderText("key.value");
    const labelInput = getByPlaceholderText(/name/i);

    fireEvent.change(keyInput, { target: { value: "test" } });
    fireEvent.change(labelInput, { target: { value: "test" } });

    await waitFor(() => expect(getByTestId("inputs-modal-confirm-button")).toBeEnabled());
  });
});
