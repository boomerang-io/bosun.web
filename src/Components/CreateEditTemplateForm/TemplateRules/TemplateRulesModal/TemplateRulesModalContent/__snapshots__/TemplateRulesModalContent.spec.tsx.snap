// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inputs --- Snapshot Test Capturing Snapshot of Inputs 1`] = `
<body
  id="app"
>
  <div>
    <form
      class="bx--bmrg-modal-form"
    >
      <div
        class="bx--modal-content"
      >
        <div
          class="bx--bmrg-text-input"
        >
          <div
            class="bx--form-item bx--text-input-wrapper"
          >
            <label
              class="bx--label"
              for="label"
            >
              <div
                class="bx--bmrg-text-input__label"
              >
                <div>
                  Name
                </div>
              </div>
            </label>
            <div
              class="bx--text-input__field-outer-wrapper"
            >
              <div
                class="bx--text-input__field-wrapper"
              >
                <input
                  class="bx--text-input bx--text__input"
                  id="label"
                  placeholder="Name"
                  title="Name"
                  type="text"
                  value="Tim Property"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="bx--bmrg-text-input"
        >
          <div
            class="bx--form-item bx--text-input-wrapper"
          >
            <label
              class="bx--label"
              for="description"
            >
              <div
                class="bx--bmrg-text-input__label"
              >
                <div>
                  Description
                </div>
              </div>
            </label>
            <div
              class="bx--text-input__field-outer-wrapper"
            >
              <div
                class="bx--text-input__field-wrapper"
              >
                <input
                  class="bx--text-input bx--text__input"
                  id="description"
                  placeholder="Description"
                  title="Description"
                  type="text"
                  value="Tim property"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="bx--bmrg-select"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="downshift-0-label"
            class="bx--list-box__wrapper"
            role="combobox"
          >
            <label
              class="bx--label"
              for="type"
              id="downshift-0-label"
            >
              <div
                style="display: flex;"
              >
                <div>
                  Type
                </div>
              </div>
            </label>
            <div
              aria-label="Choose an item"
              class="bx--combo-box bx--list-box"
            >
              <div
                aria-haspopup="true"
                aria-label="open menu"
                class="bx--list-box__field"
                data-toggle="true"
                id="type"
                role="button"
                tabindex="-1"
                type="button"
              >
                <input
                  aria-autocomplete="list"
                  aria-labelledby="downshift-0-label"
                  autocomplete="off"
                  class="bx--text-input"
                  id="type"
                  placeholder="Select an item"
                  tabindex="0"
                  type="text"
                  value="Text"
                />
                <div
                  aria-label="Clear Selection"
                  class="bx--list-box__selection"
                  role="button"
                  tabindex="0"
                  title="Clear selected item"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 32 32"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4 14.6 16 8 22.6 9.4 24 16 17.4 22.6 24 24 22.6 17.4 16 24 9.4z"
                    />
                  </svg>
                </div>
                <div
                  class="bx--list-box__menu-icon"
                >
                  <svg
                    aria-label="Open menu"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    name="chevron--down"
                    preserveAspectRatio="xMidYMid meet"
                    role="img"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                    />
                    <title>
                      Open menu
                    </title>
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="bx--bmrg-text-input"
        >
          <div
            class="bx--form-item bx--text-input-wrapper"
          >
            <label
              class="bx--label"
              for="defaultValue"
            >
              <div
                class="bx--bmrg-text-input__label"
              >
                <div>
                  Default Value
                </div>
              </div>
            </label>
            <div
              class="bx--text-input__field-outer-wrapper"
            >
              <div
                class="bx--text-input__field-wrapper"
              >
                <input
                  class="bx--text-input bx--text__input"
                  data-testid="text-input"
                  id="defaultValue"
                  placeholder="Default Value"
                  title="Default Value"
                  type="text"
                  value="dogs"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="bx--modal-footer bx--btn-set"
      >
        <button
          class="bx--btn bx--btn--secondary"
          tabindex="0"
          type="button"
        >
          Cancel
        </button>
        <button
          class="bx--btn bx--btn--primary"
          data-testid="inputs-modal-confirm-button"
          tabindex="0"
          type="submit"
        >
          Save
        </button>
      </div>
    </form>
  </div>
</body>
`;
